boa/src/builtins/array/mod.rs

```diff
@@ -1782,6 +1783,188 @@ impl Array {
+                    // iv. If fromPresent is true, then
+                    if from_present {
+                        // 1. Let fromValue be ? Get(O, from).
+                        let from_value = o.get(from, context)?;
+                        // 2. Perform ? Set(O, to, fromValue, true).
+                        o.set(to, from_value, true, context)?;
+                    // v. Else,
+                    } else {
+                        // 1. Assert: fromPresent is false.
+                        debug_assert!(!from_present);
+                        // 2. Perform ? DeletePropertyOrThrow(O, to).
+                        o.delete_property_or_throw(to, context)?;
+                    }
+                    // vi. Set k to k + 1.
+                }
+                // c. Set k to len.
+                // d. Repeat, while k > (len - actualDeleteCount + itemCount),
+                for k in ((len - actual_delete_count + item_count + 1)..=len).rev() {
+                    // i. Perform ? DeletePropertyOrThrow(O, ! ToString(𝔽(k - 1))).
+                    o.delete_property_or_throw(k - 1, context)?;
+                    // ii. Set k to k - 1.
+                }
+            }
+            // 17. Else if itemCount > actualDeleteCount, then
+            ic if ic > actual_delete_count => {
+                // a. Set k to (len - actualDeleteCount).
+                // b. Repeat, while k > actualStart,
+                for k in (actual_start + 1..=len - actual_delete_count).rev() {
+                    // i. Let from be ! ToString(𝔽(k + actualDeleteCount - 1)).
+                    let from = k + actual_delete_count - 1;
+                    // ii. Let to be ! ToString(𝔽(k + itemCount - 1)).
+                    let to = k + item_count - 1;
+                    // iii. Let fromPresent be ? HasProperty(O, from).
+                    let from_present = o.has_property(from, context)?;
+                    // iv. If fromPresent is true, then
+                    if from_present {
+                        // 1. Let fromValue be ? Get(O, from).
+                        let from_value = o.get(from, context)?;
+                        // 2. Perform ? Set(O, to, fromValue, true).
+                        o.set(to, from_value, true, context)?;
+                    // v. Else,
+                    } else {
+                        // 1. Assert: fromPresent is false.
+                        debug_assert!(!from_present);
+                        // 2. Perform ? DeletePropertyOrThrow(O, to).
+                        o.delete_property_or_throw(to, context)?;
+                    }
+                    // vi. Set k to k - 1.
+                }
+            }
+            _ => {}
+        };
+
+        // 18. Set k to actualStart.
+        let mut k = actual_start;
+
+        // 19. For each element E of items, do
+        if item_count > 0 {
+            let items = args.split_at(2).1;
+            for item in items {
```

Comment:
```suggestion
            for (idx, item) in items.iter().enumerate().map(|(i, val)| (i + actual_start, val)) {
```
This way you don't need to manually create a `k` variable and it automatically indexes all values with an offset of `actual_start`